waveshell/pkg/cirfile/cirfile.go

```diff
@@ -148,14 +150,16 @@ func CreateCirFile(fileName string, maxSize int64) (*File, error) {
 		return nil, err
 	}
 	rtn := &File{OSFile: fd, Version: CurrentVersion, MaxSize: maxSize, StartPos: FilePosEmpty}
-	err = rtn.flock(nil, syscall.LOCK_EX)
+	timeoutContext, cancelFn := context.WithTimeout(context.Background(), 20*time.Millisecond)
```

Comment:
i'm going to change this back to passing `nil`.  I think that's the right call here.  This is a new file.  The only way that the flock could block is if someone else is creating this exact same file at the same time.  In that situation it would be better to error immediately (with the EWOULDBLOCK) than to wait the 20ms and get the lock and proceed with the code.